MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -862,6 +863,9 @@ namespace WPEFramework {
                         LOGINFO("Thread joined successfully\n");
                     }
 
+                    if (!g_unsolicited_complete) {
+                        g_unsolicited_complete = true;
+                    }
                     MaintenanceManager::_instance->onMaintenanceStatusChange(notify_status);
```

Comment:
The reason why the m_statusMutex it is taken is because AS is calling startMaintenance while we are waiting for the internet connection (line 287 in task_execution_thread). The MM is still reporting IDLE at that time. When startMaintenance is called, it takes the m_statusMutex (line 1204), since MM is still in IDLE state, it is allowed to pass through the MAINTENANCE_STARTED check and this thread ends up getting blocked at line 1227 waiting for the threads to join. Meanwhile, the UNSOLICITED maintenance cycle eventually gets an internet connection and then will block on either line 192 (internet not connected) or line 300 (internet connected) since the startMaintenance thread now owns the m_stateMutex lock until the original task_execution_thread exits, which it never can.

Reply:
@sangel94403 - The current fix looks good. wanted to review why we are taking the m_statusMutex for the whole of startMaintenance & iarmEventHandler since it should be done only before updating m_notify_status. We can revisit that later.